fastify.d.ts

```diff
@@ -181,13 +182,7 @@ declare module '@fastify/error' {
   }
 }
 
-export interface ValidationResult {
-  keyword: string;
-  dataPath: string;
-  schemaPath: string;
-  params: Record<string, string | string[]>;
-  message: string;
-}
+export interface ValidationResult extends AjvErrorObject {}
```

Comment:
The validation system is pluggable.
While I think this is a fix for the default settings, we might want something slightly different.

wdyt @Eomm?

Reply:
That's correct. Thanks @mcollina for pointing out.

If we are to make `ValidationResult` generic (as the validation system is pluggable),

then we probably also need to make [`FastifyError`](https://github.com/fastify/fastify/blob/main/fastify.d.ts#L179) generic,

because `FastifyError` depends on `ValidationResult`.

This looks like a bigger change than what the original issue describes.

In this case, I would suggest to close this pull request and open up a new issue to make `ValidationResult` generic.

Let me know what you think. Thanks!

Reply:
Definitely!
Maybe you can send a PR to just update the old/new properties as hot fix while you work on the new PR?

Reply:
Sure! Here is the hotfix PR #4070